src/machine/machine_rp2040_i2c.go

```diff
@@ -73,11 +82,27 @@ var (
 //
 // Performs only a write transfer.
 func (i2c *I2C) Tx(addr uint16, w, r []byte) error {
+	if i2c.mode != I2CModeController {
+		return ErrI2CWrongMode
+	}
+
 	// timeout in microseconds.
 	const timeout = 40 * 1000 // 40ms is a reasonable time for a real-time system.
 	return i2c.tx(uint8(addr), w, r, timeout)
 }
 
+// Listen starts listening for I2C requests sent to specified address
+//
+// addr is the address to listen to
+// maxRecv controls the receive buffer size
+func (i2c *I2C) Listen(addr uint16, maxRecv uint32) error {
```

Comment:
Is maxRecv redundant with the buffer passed into `WaitForEvent`?

Reply:
It’s needed in the API for nrf chipsets where a fixed size receive buffer is passed to the peripheral for receiving messages.

Reply:
I've taken another look - agree that parameter can go.